recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -601,6 +638,17 @@ def main(argv):
         ====================================
 Hit any Key to Exit:
         '''
+        # Rollback the firmware when the upgrade fails
+        try:
+            with open('/root/.rollback_fw/iot2050-image-boot.bin', 'rb') as f:
+                print("Rollback to the previous firmware")
+                updater = FirmwareUpdate(f)
+                updater.update_firmware()
+        except (IOError, UpgradeError):
+            print("**********************************")
+            print("Rollback firmware fails")
+            print("**********************************")
+
```

Comment:
At least should before `intput_reminder = '''`

Reply:
Do not use `/root/.rollback_fw/iot2050-image-boot.bin` directly, use some method to get the backed up firmwares from the instance of `ManagedFirmwareUpdate`